src/components/swapv2/PairSuggestion/ListPair.tsx

```diff
@@ -0,0 +1,154 @@
+        <Trans>{text}</Trans>
+      </Text>
+    </Flex>
+  </Container>
+)
+
+export default function ListPair({
+  isShowListPair,
+  hasShadow,
+  suggestedAmount,
+  favoritePairs,
+  suggestedPairs,
+  isSearch,
+  selectedIndex,
+  onSelectPair,
+  onClickStar,
+}: {
+  suggestedAmount: string
+  hasShadow?: boolean
+  selectedIndex: number
+  isShowListPair: boolean
+  favoritePairs: SuggestionPairData[]
+  suggestedPairs: SuggestionPairData[]
+  isSearch: boolean
+  onSelectPair: (item: SuggestionPairData) => void
+  onClickStar: (item: SuggestionPairData) => void
+}) {
+  const theme = useTheme()
+  const { account } = useActiveWeb3React()
+  const isShowNotfound = isSearch && !suggestedPairs.length && !favoritePairs.length
+  const isShowNotfoundFavoritePair = !favoritePairs.length && !isSearch
+  return isShowListPair ? (
+    <MenuFlyout showList={isShowListPair} tabIndex={0} className="no-blur" hasShadow={hasShadow}>
+      {isShowNotfound && (
+        <TextWithIcon
+          color={theme.subText}
+          text="We could not find anything. Try again."
+          icon={<AlertTriangle color={theme.subText} size={17} />}
+        />
+      )}
+      {account && (
+        <>
+          {favoritePairs.length > 0 && !isSearch && (
+            <Container>
+              <Title>
+                <Flex justifyContent="space-between">
+                  <Trans>Favourites</Trans>
+                  <div>
+                    {favoritePairs.length}/{MAX_FAVORITE_PAIRS}
+                  </div>
+                </Flex>
+              </Title>
+            </Container>
+          )}
+          {isShowNotfoundFavoritePair && (
+            <TextWithIcon
+              color={theme.subText}
+              icon={<Star color={theme.text3} size={20} />}
+              text={'Your favourite pairs will appear here'}
+            />
+          )}
+          {favoritePairs.map((item, i) => (
+            <PairSuggestionItem
+              onSelectPair={() => onSelectPair(item)}
+              onClickStar={() => onClickStar(item)}
+              amount={suggestedAmount}
+              isActive={selectedIndex === i}
+              data={item}
+              isFavorite
+              key={item.tokenIn + item.tokenOut + i}
+            />
+          ))}
+          {!isSearch && <Break />}
+        </>
+      )}
+      {suggestedPairs.length > 0 && (
+        <>
+          {!isSearch && (
+            <Container>
+              <Title>
+                <Trans>Top traded pairs</Trans>
+              </Title>
+            </Container>
+          )}
+          {suggestedPairs.map((item, i) => (
+            <PairSuggestionItem
+              onSelectPair={() => onSelectPair(item)}
+              onClickStar={() => onClickStar(item)}
+              amount={suggestedAmount}
+              isActive={selectedIndex === favoritePairs.length + i}
+              data={item}
+              key={item.tokenIn + item.tokenOut + i}
```

Comment:
Isn't just `i` enough?

Reply:
some time it's not good.
In this case,   list update very quickly,    I don't think that use "index" here is good

Reply:
I don't see the differenece between `key={i}` and `key={item.tokenIn + item.tokenOut + i}`. Same stuff bro.

Reply:
My question should be: `i` and yours are the same. So maybe remove `i`.

Reply:
yes Will remove ``i`` .